### ISSUE TITLE
MediaLibrary.getAssetsAsync returns unusable URIs on Samsung devices

### ISSUE THREAD

**Issue by brandonpearcy** [expo/expo#18569]
### Summary

When fetching media assets with expo-media-library's getAssetsAsync, the returned URIs are not usable for viewing or reading image files on some popular Samsung devices.

Here is an example photo asset from a freshly factory reset Samsung Galaxy S9:
` DEBUG  {"asset": {"albumId": "-1739773001", "creationTime": 1659992721680, "duration": 0, "filename": "20220808_170521.jpg", "height": 4032, "id": "26", "mediaType": "photo", "modificationTime": 1659992721000, "uri": "file:///storage/emulated/0/DCIM/Camera/20220808_170521.jpg", "width": 3024}}
`

When using the URI in a vanilla React Native Image component, I would expect the photo to render correctly, as it does on iOS and many other Android phones (e.g. Google Pixel 5). However, the image is empty. When trying to read these files (e.g. to upload them to a server) an `EACCES: permission denied` error is thrown. I've ensured that all Storage permissions are indeed granted, according to a variety of permissions library including `expo-media-library` and `react-native-permissions`.

```
  const [status] = MediaLibrary.usePermissions()
  console.debug({status})
  DEBUG  {"status": {"canAskAgain": true, "expires": "never", "granted": true, "status": "granted"}}
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.3.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.19.3 - /opt/homebrew/opt/node@14/bin/node
      Yarn: 1.22.18 - /opt/homebrew/bin/yarn
      npm: 6.14.17 - /opt/homebrew/opt/node@14/bin/npm
      Watchman: 2022.06.06.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8512546
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    npmPackages:
      react-native: 0.68.2 => 0.68.2 
    npmGlobalPackages:
      expo-cli: 5.4.9
    Expo Workflow: managed

### Reproducible demo

In order to reproduce this issue, you will need to run the following React Native app on a physical Samsung device, such as a Samsung Galaxy S9 or Galaxy Note 9.

AndroidManifest.xml
=======
```xml
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
```

App.tsx
======
```js
import * as MediaLibrary from 'expo-media-library'
import React, {useEffect, useState} from 'react'
import {Image, StyleSheet, View} from 'react-native'

export const App = () => {
  const [mediaUrl, setMediaUrl] = useState('')

  useEffect(() => {
    ;(async () => {
      await MediaLibrary.getPermissionsAsync()
      const pagedInfo = await MediaLibrary.getAssetsAsync({
        first: 1,
        mediaType: ['photo'],
        sortBy: 'modificationTime',
      })
      const uri = pagedInfo.assets[0]?.uri || ''
      console.debug({uri})
      setMediaUrl(uri)
    })()
  }, [])

  return (
    <View>
      <Image
        source={{
          uri: 'https://image.shutterstock.com/image-photo/example-word-written-on-wooden-260nw-1765482248.jpg',
        }}
        style={styles.image}
      />
      {mediaUrl ? (
        <Image
          source={{
            uri: mediaUrl,
          }}
          style={styles.image}
        />
      ) : null}
    </View>
  )
}

const styles = StyleSheet.create({
  image: {backgroundColor: 'pink', width: 200, height: 200},
})
```

**Comment by brentvatne** [comment]
@lukmccall - do you have any insight on this?

**Comment by lukmccall** [comment]
@brentvatne, unfortunately, no :/ But it seems like a bug in Samsung software. It seems to be working on OnePlus and Google Pixel.

**Comment by brandonpearcy** [comment]
I was able to track down a possible solution. Samsung devices are still running Android 10. When I added            
 `android:requestLegacyExternalStorage="true"` to `AndroidManifest.xml` and rebuilt the app, expo-media-library was able to successfully read and write media files. I don't know if there's a way to make expo-media-library work seamlessly for Android 29 and below but—if not—it might just be enough to add a little documentation explaining the workaround.
